### PR TITLE
implement model classes following MVC pattern for Spring Boot backend

### PR DESCRIPTION
**Pull request by NkwaTambe**

### CODE REVIEW

File: power-pay-backend/src/main/java/com/adorsys/gis/powerpay/powerpaybackend/domain/CheckBalance.java [hunk 1]
```diff
@@ -0,0 +1,5 @@
+package com.adorsys.gis.powerpay.powerpaybackend.domain;
+
+public class CheckBalance extends Procedure {
```
Comment (stephane-segning) [inline power-pay-backend/src/main/java/com/adorsys/gis/powerpay/powerpaybackend/domain/CheckBalance.java:3]: This one should also be an entity na?

File: power-pay-backend/src/main/java/com/adorsys/gis/powerpay/powerpaybackend/domain/Procedure.java [hunk 2]
```diff
@@ -0,0 +1,71 @@
+package com.adorsys.gis.powerpay.powerpaybackend.domain;
+import jakarta.persistence.Entity;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+
+@Entity
+public class Procedure {
```
Comment (stephane-segning) [inline power-pay-backend/src/main/java/com/adorsys/gis/powerpay/powerpaybackend/domain/Procedure.java:8]: Shouldn't this one be an abstract class?

File: power-pay-backend/src/main/java/com/adorsys/gis/powerpay/powerpaybackend/domain/Procedure.java [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+        this.phoneNumber = phoneNumber;
+    }
+
+    public enum Status {
+        DONE,
+        WAITING,
+        ERROR
+    }
+
+    public enum Type {
```
Comment (stephane-segning) [inline power-pay-backend/src/main/java/com/adorsys/gis/powerpay/powerpaybackend/domain/Procedure.java:67]: Put this in a separate file please

File: power-pay-backend/src/main/java/com/adorsys/gis/powerpay/powerpaybackend/domain/Procedure.java [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+
+    public String getPhoneNumber() {
+        return phoneNumber;
+    }
+
+    public void setPhoneNumber(String phoneNumber) {
+        this.phoneNumber = phoneNumber;
+    }
+
+    public enum Status {
```
Comment (stephane-segning) [inline power-pay-backend/src/main/java/com/adorsys/gis/powerpay/powerpaybackend/domain/Procedure.java:61]: Put this in a separate file please

**Review by stephane-segning — CHANGES_REQUESTED** [review]
Good work @NkwaTambe. Please address these small points so that we could close this ticket

File: power-pay-backend/src/main/java/com/adorsys/gis/powerpay/powerpaybackend/domain/CheckBalance.java (hunk 1 above)
Comment (NkwaTambe) [reply power-pay-backend/src/main/java/com/adorsys/gis/powerpay/powerpaybackend/domain/CheckBalance.java:3]: ok sir

File: power-pay-backend/src/main/java/com/adorsys/gis/powerpay/powerpaybackend/domain/Procedure.java (hunk 4 above)
Comment (NkwaTambe) [reply power-pay-backend/src/main/java/com/adorsys/gis/powerpay/powerpaybackend/domain/Procedure.java:61]: In general i should put the two enumerations in a different file(same file for both enumerations ) ???

Comment (stephane-segning) [reply power-pay-backend/src/main/java/com/adorsys/gis/powerpay/powerpaybackend/domain/Procedure.java:61]: Each enum should be in a different file

Comment (NkwaTambe) [reply power-pay-backend/src/main/java/com/adorsys/gis/powerpay/powerpaybackend/domain/Procedure.java:61]: ok sir

Comment (bryandino673) [reply power-pay-backend/src/main/java/com/adorsys/gis/powerpay/powerpaybackend/domain/Procedure.java:61]: ok sir

File: power-pay-backend/src/main/java/com/adorsys/gis/powerpay/powerpaybackend/domain/CheckBalance.java (hunk 1 above)
Comment (bryandino673) [reply power-pay-backend/src/main/java/com/adorsys/gis/powerpay/powerpaybackend/domain/CheckBalance.java:3]: ok sir

File: power-pay-backend/src/main/java/com/adorsys/gis/powerpay/powerpaybackend/domain/Procedure.java (hunk 4 above)
Comment (NkwaTambe) [reply power-pay-backend/src/main/java/com/adorsys/gis/powerpay/powerpaybackend/domain/Procedure.java:61]: Hello sir i just did some research on how to place enum's on spring boot application..

i notice i need to create a package called enum's then create two classes inside to put the "Status" and the "Type" enums

Comment (stephane-segning) [reply power-pay-backend/src/main/java/com/adorsys/gis/powerpay/powerpaybackend/domain/Procedure.java:61]: You already have a package named domain. So you'll want to put the Type and Status class there. As there might be multiple other Status & Type in the future, it's better to use names like `ProcedureStatus`.

Comment (NkwaTambe) [reply power-pay-backend/src/main/java/com/adorsys/gis/powerpay/powerpaybackend/domain/Procedure.java:61]: ok sir, understood

**Review by stephane-segning — APPROVED** [review]
Awesome! Thank you @NkwaTambe @bryandino673

**Pull request merged**